prog/encodingexec.go

```diff
@@ -54,8 +64,9 @@ func (p *Prog) SerializeForExec(buffer []byte, pid int) error {
 		foreachArg(c, func(arg, _ *Arg, _ *[]*Arg) {
 			if arg.Kind == ArgPointer && arg.Res != nil {
 				foreachSubargOffset(arg.Res, func(arg1 *Arg, offset uintptr) {
-					if len(arg1.Uses) != 0 {
-						w.args[arg1] = argInfo{Offset: offset}
+					w.args[arg1] = argInfo{Addr: physicalAddr(arg) + offset}
```

Comment:
This generates tons of garbage and slows down execution. I especially added `if len(arg1.Uses) != 0 {`.
Can we also collect all args that are used by csums into a separate map during csum constructions, and then here do:
```
if len(arg1.Uses) != 0 || csumSourceArg[arg1] {
    w.args[arg1] = ...
}
```
?